components/constellation/constellation.rs

```diff
@@ -5396,24 +5472,25 @@ where
     }
 
     /// Send the frame tree for the given webview to the compositor.
-    fn update_frame_tree_if_focused(
-        &mut self,
-        top_level_browsing_context_id: TopLevelBrowsingContextId,
-    ) {
-        // Only send the frame tree if the given webview is focused.
-        if let Some(focused_webview_id) = self.webviews.focused_webview().map(|(id, _)| id) {
-            if top_level_browsing_context_id != focused_webview_id {
-                return;
+    fn update_webview(&mut self, top_level_browsing_context_id: TopLevelBrowsingContextId) {
+        if !cfg!(feature = "multiview") {
+            if let Some(focused_webview_id) = self.webviews.focused_webview().map(|(id, _)| id) {
+                if top_level_browsing_context_id != focused_webview_id {
+                    return;
+                }
+            } else {
+                self.webviews.focus(top_level_browsing_context_id);
             }
         }
```

Comment:
I'm a bit confused as to why this only happens in the case that multiview is disabled. In any case, I think this should be using `any()`.

Reply:
i agree that this is confusing, i think this was a combination of the way the patch evolved ([1](https://github.com/servo/servo/pull/30648/commits/6cd285f51644e3abc8b046dd17730d5762049644#diff-95aa2eb890fe794918afdadfb6a89e026ec80222ceb8e5a21768d528e9ce036c), [2](https://github.com/servo/servo/pull/30648/commits/510c6d4215d20b379d38ad9617fb20b658f65e2e#diff-95aa2eb890fe794918afdadfb6a89e026ec80222ceb8e5a21768d528e9ce036c)) and erring on the side of caution ([3](https://github.com/servo/servo/pull/30648/commits/f6403326bee652fa6cc84de75cb6bfa92f446c1f#diff-95aa2eb890fe794918afdadfb6a89e026ec80222ceb8e5a21768d528e9ce036c)), but then again the version we landed was unconditional ([4](https://github.com/servo/servo/pull/30842/commits/d3ce4a7280ea42555fab96385f568d42f8d6cb4d#diff-95aa2eb890fe794918afdadfb6a89e026ec80222ceb8e5a21768d528e9ce036cL5391-R5453)). i’ll look into it :)

Reply:
305a50b9e632382b434faec61dce9abf7d0b6b01 removes both the feature gate and the focus check itself, hopefully the tests still pass!